core/src/com/punchbrain/awake/model/Circuit.java

```diff
@@ -23,28 +29,57 @@ public Circuit(GameObject lamp, GameObject flipSwitch, Light lightObject, AudioM
         this.audioManager = audioManager;
     }
 
-    public boolean isOn(){
-        return isOn;
+
+    public Circuit updatePassiveBehaviour(float delta) {
+        updateState(delta);
+        updateAnimationState();
+        if (state == State.ON) {
+            this.lightObject.setDistance(lightRadius);
+        } else {
+            this.lightObject.setDistance(0);
+        }
+        return this;
     }
 
-    public Light getLightObject(){
-        return lightObject;
+    public Circuit resolvePlayerCollision() {
+        // TODO: add player 'push' conditional
+        this.flipSwitch();
+        return this;
     }
 
-    public void flipSwitch(){
-        isOn = !isOn;
-        if (isOn) {
-            audioManager.spawnSound(Assets.Sounds.SWITCH_ON, flipSwitch.getLeft(), flipSwitch.getTop(), 1f, 1f, 200f);
+    private void updateAnimationState() {
+        if (this.state == State.ON) {
+            this.flipSwitch.setAttribute(LampState.class, LampState.ON);
+            this.lamp.setAttribute(LampState.class, LampState.ON);
         } else {
-            audioManager.spawnSound(Assets.Sounds.SWITCH_OFF, flipSwitch.getLeft(), flipSwitch.getTop(), 1f, 1f, 200f);
+            this.flipSwitch.setAttribute(LampState.class, LampState.OFF);
+            this.lamp.setAttribute(LampState.class, LampState.OFF);
         }
     }
 
-    public GameObject getLampGameObject(){
+    public GameObject getLampGameObject() {
         return lamp;
     }
 
-    public GameObject getFlipSwitchGameObject(){
+    public GameObject getFlipSwitchGameObject() {
         return flipSwitch;
     }
+
+    private void updateState(float delta) {
+        if (state == State.ON) {
+            deltaAccumulator += delta;
+            if (deltaAccumulator >= deltaLimit) {
+                state = State.BROKEN;
+            }
+        }
+    }
+
+    private void flipSwitch() {
+        if (state == State.OFF) {
+            state = State.ON;
+            audioManager.spawnSound(Assets.Sounds.SWITCH_ON, flipSwitch.getLeft(), flipSwitch.getTop(), 1f, 1f, 200f);
```

Comment:
removed the off since for timed ones we don't want to be able to.
Could add sound affect for when the state is broken so that you can hear it click but the lighting doesn't change.